tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -4406,6 +4405,202 @@ TEST_F(InputProcessorFixture, epJSONgetObjectItem_minfields)
+    alphaFieldValue = state->dataInputProcessing->inputProcessor->getAlphaFieldValue(*state, obj_type2, mat1, objectSchemaProps, "roughness");
+    EXPECT_EQ(alphaFieldValue, "MEDIUMROUGH");
+    numericFieldValue = state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type2, mat1, objectSchemaProps, "thickness");
+    EXPECT_NEAR(numericFieldValue, 0.2, 0.0001);
+    numericFieldValue =
+        state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type2, mat1, objectSchemaProps, "solar_absorptance");
+    EXPECT_NEAR(numericFieldValue, 0.5, 0.0001);
+    // Defaults from schema
+    numericFieldValue =
+        state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type2, mat1, objectSchemaProps, "thermal_absorptance");
+    EXPECT_NEAR(numericFieldValue, 0.9, 0.0001);
+    // Fields beyond min-fields also return their default if they have one (unlike getObjectItem)
+    numericFieldValue =
+        state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type2, mat1, objectSchemaProps, "visible_absorptance");
+    EXPECT_NEAR(numericFieldValue, 0.7, 0.0001);
+    // or zero if they don't have a default (in this case it's a required field, so it would have failed before now)
+    numericFieldValue = state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type2, mat1, objectSchemaProps, "specific_heat");
+    EXPECT_NEAR(numericFieldValue, 0.0, 0.0001);
+}
+
+TEST_F(InputProcessorFixture, epJSONgetFieldValue_AutosizefromJSON)
+{
+
+    json root;
+    std::string obj_type1 = "WaterHeater:Mixed";
+    std::string name1 = "Water Heater 1";
+    json wh1 = {{"tank_volume", "autosize"}, {"heater_maximum_capacity", "nonsense"}, {"source_side_effectiveness", "nothing"}};
+    EXPECT_TRUE(wh1.is_object());
+    root[obj_type1][name1] = wh1;
+
+    state->dataInputProcessing->inputProcessor->epJSON = root;
+    state->dataGlobal->isEpJSON = true;
+    state->dataInputProcessing->inputProcessor->initializeMaps();
+    std::string alphaFieldValue;
+    Real64 numericFieldValue = 0.0;
+    json objectSchemaProps;
+
+    // Water heater object
+    objectSchemaProps = state->dataInputProcessing->inputProcessor->getObjectSchemaProps(*state, obj_type1);
+    // User inputs from above
+    // If the field is autosizable and alpha input will return -99999
+    numericFieldValue = state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type1, wh1, objectSchemaProps, "tank_volume");
+    EXPECT_EQ(numericFieldValue, -99999);
+    numericFieldValue =
+        state->dataInputProcessing->inputProcessor->getRealFieldValue(*state, obj_type1, wh1, objectSchemaProps, "heater_maximum_capacity");
+    EXPECT_EQ(numericFieldValue, -99999);
+    // Even a field that is not autoszable will return -99999 here (assuming that gets checked upon epJSON input processing)
```

Comment:
I guess it is safe to assume this will always be caught. I wonder if this is actually unit tested somewhere.

    ** Severe  ** <root>[Building][Building][loads_convergence_tolerance_value] - Value type "string" for input "autosize" not permitted by 'type' constraint.

Reply:
Not sure why it says this input field type is "constraint"?

    "loads_convergence_tolerance_value": {
        "type": "number",